### DISCUSSION TITLE
How to add an Object to an Array?

### DISCUSSION THREAD

**Discussion by MaybeELEMENT** [nlohmann/json#2985] [Q&A] [answered]
Hello, I have a question. how can I add an object to an array?
so I have this JSON script:
```json
[]
```
I want to make it to this:
```json
[
{"label": "key", "value": "var"},
{"label": "string", "value": "hello world"}
]
```
But how can I do it?
I tried
```cpp
res.push_back({json::object({"label", "key", "value", "var"})});
res.push_back({json::object({"label", "string", "value", "hello world"})});
```
But it just give me error:
`libc++abi: terminating with uncaught exception of type nlohmann::detail::type_error: [json.exception.type_error.301] cannot create object from initializer list`

**Comment by nlohmann** [comment, marked as answer]
The `object` function expects a list of pairs (see https://json.nlohmann.me/api/basic_json/object/). You could also use the second version of `push_back` and provide an object value directly (see https://json.nlohmann.me/api/basic_json/push_back/).

For your example:

```cpp
res.push_back(json::object({{"label", "key"}, {"value", "var"}}));
res.push_back(json::object({{"label", "string"}, {"value", "hello world"}}));
```

You can also skip the `json::object` here, because list of pairs are treated as object anyway:

```cpp
res.push_back({{"label", "key"}, {"value", "var"}});
res.push_back({{"label", "string"}, {"value", "hello world"}});
```
